src/d3a/d3a_core/util.py

```diff
@@ -332,16 +331,22 @@ def update_nested_settings(class_object, class_name, settings_dict):
         update_nested_settings(setting_class, settings_class_name, advanced_settings)
 
 
-def generate_market_slot_list(area=None):
-    """
-    Returns a list of all slot times
-    """
-    config = GlobalConfig if area is None else area.config
-    if not hasattr(config, 'market_slot_list') or len(config.market_slot_list) == 0:
-        config.market_slot_list = generate_market_slot_list_from_config(
-            config.sim_duration, config.start_date, config.market_count, config.slot_length
-        )
-    return config.market_slot_list
+def generate_market_slot_list(start_date=None, time_span=None):
+    if start_date is None:
+        start_date = GlobalConfig.start_date
+    if time_span is None:
+        time_span = GlobalConfig.sim_duration
+    sim_duration_plus_future_markets = time_span + GlobalConfig.slot_length * \
+        (GlobalConfig.market_count - 1)
```

Comment:
Can you help me understand the reason for one subtraction offset here?

Reply:
yes, 
if `GlobalConfig.market_count=1`, the last market slot should be: `23:45`
if `GlobalConfig.market_count=2`, the last market slot should be: `00:00` (next day)
ans so on. If you add another slot to `time_span` there will be an offset of one market slot that is not wanted here.